ocrd/ocrd/resource_manager.py

```diff
@@ -210,16 +213,31 @@ def _download_impl(self, url, filename, progress_cb=None, size=None):
 
     def _copy_impl(self, src_filename, filename, progress_cb=None):
         log = getLogger('ocrd.resource_manager._copy_impl')
-        log.info("Copying %s" % src_filename)
-        with open(filename, 'wb') as f_out, open(src_filename, 'rb') as f_in:
-            while True:
-                chunk = f_in.read(4096)
-                if chunk:
-                    f_out.write(chunk)
-                    if progress_cb:
-                        progress_cb(len(chunk))
-                else:
-                    break
+        log.info("Copying %s to %s", src_filename, filename)
+        if Path(src_filename).is_dir():
+            log.info(f"Copying recursively from {src_filename} to {filename}")
+            for child in Path(src_filename).rglob('*'):
+                child_dst = Path(filename) / child.relative_to(src_filename)
+                child_dst.parent.mkdir(parents=True, exist_ok=True)
+                with open(child_dst, 'wb') as f_out, open(child, 'rb') as f_in:
```

Comment:
That's not recursive, though: If the source directory contains subdirectories, this fails:
```
ocrd resmgr download -t directory -n ocrd_anybaseocr/seg_model ocrd-anybaseocr-tiseg seg_model
02:48:53.405 INFO ocrd.cli.resmgr - No resources found in registry
02:48:53.405 INFO ocrd.cli.resmgr - Copying unregistered resource 'seg_model' (ocrd_anybaseocr/seg_model)
02:48:53.429 INFO ocrd.resource_manager._copy_impl - Copying /ocrd_all/ocrd_anybaseocr/seg_model to /.local/share/ocrd-resources/ocrd-anybaseocr-tiseg/seg_model
02:48:53.429 INFO ocrd.resource_manager._copy_impl - Copying recursively from /ocrd_all/ocrd_anybaseocr/seg_model to /.local/share/ocrd-resources/ocrd-anybaseocr-tiseg/seg_model
Traceback (most recent call last):
  File "/bin/ocrd", line 8, in <module>
    sys.exit(cli())
  File "/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/lib/python3.6/site-packages/ocrd/cli/resmgr.py", line 146, in download
    progress_cb=lambda delta: bar.update(delta)
  File "/lib/python3.6/site-packages/ocrd/resource_manager.py", line 274, in download
    self._copy_impl(url, fpath, progress_cb)
  File "/lib/python3.6/site-packages/ocrd/resource_manager.py", line 222, in _copy_impl
    with open(child_dst, 'wb') as f_out, open(child, 'rb') as f_in:
IsADirectoryError: [Errno 21] Is a directory: '/ocrd_all/ocrd_anybaseocr/seg_model/variables'
```

IMO the only way to make this work is by encapsulating both conditional branches into a single function that can call itself recursively.